dvc/commands/update.py

```diff
@@ -55,6 +56,12 @@ def add_parser(subparsers, parent_parser):
         default=False,
         help="Update all stages in the specified directory.",
     )
+    update_parser.add_argument(
+        "--check-only",
+        action="store_true",
+        default=False,
+        help="Check if data is up-to-date without actually downloading it.",
+    )
```

Comment:
WDYT about `--dry-run` (we use it elsewhere too)?

Reply:
Thinking about it, even `--check-only` is misleading. I renamed the flag to `--no-download` to align it with `import-url --no-download`. Do you think it could be worth it to have a `--dry-run` option too?

Reply:
Is `--check-only` still misleading? Seems like that's the clearest differentiator from `--no-exec`.